### ISSUE TITLE
Type intersection allows comparison with string and number values

### ISSUE THREAD

**Issue by BenoitZugmeyer** [microsoft/TypeScript#45489]
# Bug Report

### 🔎 Search Terms

intersection comparison greater lesser

### 🕗 Version & Regression Information

Based on the typescript playground versions, the issue occurs with `4.3.5` and `4.4.0-beta`, but not with `4.2.3` and below.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.3.5#code/CYUwxgNghgTiAEEQBd4DMD2GBc8De8UuAjAL4CwAUGBgHYDOGSAdBBgOYAUmG8AfPADMASngB6MYXjIAngAcEIGDAwx4AS3rw4c1chDBCWkAA8FYfcCpVQkWAiSoeAdXXIAFgElguAkXjE8KTwAGT4Gj7wtACuALYARkpBVDQMTCCsHNxYrh7e-EKiElG8SipqmtogujCWQA)

### 💻 Code

```ts
declare let foo: { a: 1}
console.log(foo > 3) // a type error is reported as expected

declare let fooWithId: { a: 1 } & { id: number }
console.log(fooWithId > 3) // no error is reported
```

### 🙁 Actual behavior

No error is reported when comparing a value of type produced by an intersection and a string or a number.

### 🙂 Expected behavior

An error should be reported, since comparing an object and a number is not generally useful and probably an issue.

**Comment by a-tarasyuk** [comment]
It seems this issue was fixed. [Nightly Build](https://www.typescriptlang.org/play?ts=4.5.0-dev.20210910#code/CYUwxgNghgTiAEEQBd4DMD2GBc8De8UuAjAL4CwAUGBgHYDOGSAdBBgOYAUmG8AfPADMASngB6MYXjIAngAcEIGDAwx4AS3rw4c1chDBCWkAA8FYfcCpVQkWAiSoeAdXXIAFgElguAkXjE8KTwAGT4Gj7wtACuALYARkpBVDQMTCCsHNxYrh7e-EKiElG8SipqmtogujCWQA) reports errors for two cases. 

/cc @andrewbranch

**Comment by andrewbranch** [comment]
I’ll let @armanio123 add a passing test and close it then 👍
